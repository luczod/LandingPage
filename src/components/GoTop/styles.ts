import styled, { DefaultTheme, css } from "styled-components";

export const Container = styled.a`
  ${({ theme }) => css`
    background: ${theme.colors.primaryColor};
    color: ${theme.colors.white};
    position: fixed;
    right: 3rem;
    bottom: 2rem;
    z-index: 6;
    display: flex;
    align-items: center;
    justify-content: center;
    height: 4rem;
    width: 4rem;
    opacity: 0.8;
  `}
`;
