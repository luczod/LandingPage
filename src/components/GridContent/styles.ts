import styled, { DefaultTheme, css } from "styled-components";

export const Container = styled.div`
  ${({ theme }) => css`
    text-align: center;
    padding: 0;
    margin: 0 auto;
  `}
`;

export const HtmlWrap = styled.div`
  ${({ theme }) => css`
    margin: 0;
  `}
`;
