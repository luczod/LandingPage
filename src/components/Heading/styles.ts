import styled, { DefaultTheme, css } from "styled-components";
type OnlyKeys = keyof typeof titleSize;
interface ISProps {
  $colorark: boolean;
  size: OnlyKeys;
  $uppertitle?: boolean;
}

const titleSize = {
  small: (theme: DefaultTheme) => css`
    font-size: ${theme.font.sizes.md};
  `,
  medium: (theme: DefaultTheme) => css`
    font-size: ${theme.font.sizes.lg};
  `,
  big: (theme: DefaultTheme) => css`
    font-size: ${theme.font.sizes.xl};
  `,
  huge: (theme: DefaultTheme) => css`
    font-size: ${theme.font.sizes.xhg};
    ${mediaFont(theme)};
  `,
};

const mediaFont = (theme: DefaultTheme) => css`
  @media ${theme.media.lteMedium} {
    font-size: ${theme.font.sizes.xl};
  }
`;

export const Title = styled.h1<ISProps>`
  ${({ theme, size, $uppertitle, $colorark: colorDark }) => css`
    color: ${colorDark ? theme.colors.primaryColor : theme.colors.white};
    ${titleSize[size](theme)};
    text-transform: ${$uppertitle ? "uppercase" : "none"};
  `}
`;
