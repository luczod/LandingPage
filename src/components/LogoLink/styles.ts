import styled, { css } from "styled-components";

export const Container = styled.a`
  ${({ theme }) => css`
    color: inherit;
    display: flex;
    align-items: center;
    > img {
      max-height: 2.5rem;
    }
  `}
`;
