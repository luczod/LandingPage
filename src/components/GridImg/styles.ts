import styled, { css } from "styled-components";
import { Container as TextComponent } from "../Textfn/styles";

interface IPros {
  open?: boolean;
}

export const Container = styled.div`
  ${({ theme }) => css`
    > ${TextComponent} {
      margin-bottom: ${theme.spacings.lg};
    }
  `}
`;

export const Grid = styled.div`
  ${({ theme }) => css`
    counter-reset: grid-counter;
    display: grid;
    grid-template-columns: repeat(auto-fill, minmax(280px, 1fr));
    gap: ${theme.spacings.lg};
  `}
`;

export const GridItem = styled.div`
  ${({ theme }) => css`
    overflow: hidden;
    border-radius: 1rem;
  `}
`;

export const Image = styled.img`
  ${({ theme }) => css`
    width: 100%;
    transition: all 300ms ease-in-out;

    &:hover {
      transform: scale(1.2) rotate(10deg);
    }
  `}
`;

export const ModalImg = styled.div`
  ${({ theme }) => css`
    display: none; /* Hidden by default */
    position: fixed; /* Stay in place */
    z-index: 1; /* Sit on top */
    padding-top: 100px; /* Location of the box */
    left: 0;
    top: 0;
    width: 100%;
    height: 100%;
    background-color: rgba(0 0 0 / 0.3); /* color-Black w/ opacity */
    border-radius: 1rem;

    & > span {
      position: fixed;
      cursor: pointer;
      top: 4rem;
      right: 2rem;
      color: ${theme.colors.secondaryColor};
      font-size: ${theme.font.sizes.hg};
      &::hover {
        color: ${theme.colors.white};
      }
      @media only screen and (max-width: 768px) {
        left: 1rem;
      }
    }

    & > img {
      margin: auto;
      display: block;
      width: 80%;
      max-width: 700px;
      border-radius: 1rem;
      /* 100% Image Width on Smaller Screens */
      @media only screen and (max-width: 768px) {
        width: 100%;
        padding: 0 0.8rem;
      }
    }
    & > div {
      margin: auto;
      display: block;
      width: 80%;
      max-width: 700px;
      text-align: center;
      font-size: ${theme.font.sizes.md};
      color: ${theme.colors.white};
      padding: 10px 0;
      height: 150px;
    }
  `}
`;
