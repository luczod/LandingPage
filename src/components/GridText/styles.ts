import styled, { css } from "styled-components";
import { Title as ItemTitle } from "../Heading/styles";
import { Container as TextComponent } from "../Textfn/styles";

export const Container = styled.div`
  ${({ theme }) => css`
    ${TextComponent} {
      margin-bottom: ${theme.spacings.lg};
    }
  `}
`;

export const Grid = styled.div`
  ${({ theme }) => css`
    counter-reset: grid-counter;
    display: grid;
    grid-template-columns: repeat(auto-fill, minmax(280px, 1fr));
    gap: ${theme.spacings.lg};
    overflow: hidden;
    width: 100%;
    @media ${theme.media.lteMedium} {
      grid-template-columns: 1fr;
    }
  `}
`;

export const GridItem = styled.article`
  ${({ theme }) => css`
    ${ItemTitle} {
      position: relative;
      left: 5rem;
    }

    ${ItemTitle}::before {
      counter-increment: grid-counter;
      content: counter(grid-counter);
      position: absolute;
      font-size: ${theme.font.sizes.xhg};
      top: -2rem;
      left: -5rem;
      transform: rotate(8deg);
    }
  `}
`;
