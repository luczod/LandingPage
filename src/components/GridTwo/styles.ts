import styled, { DefaultTheme, css } from "styled-components";
import { Title } from "../Heading/styles";

interface IProps {
  background: boolean;
}

export const Container = styled.div<IProps>`
  ${({ theme, background }) => css`
    display: grid;
    grid-template-columns: 1fr 2fr;
    align-items: center;
    gap: ${theme.spacings.lg};

    @media ${theme.media.lteMedium} {
      grid-template-columns: 1fr;
    }

    ${Title} {
      margin-bottom: ${theme.spacings.xl};
    }
  `}
`;

export const TextColumn = styled.div`
  ${({ theme }) => css`
    @media ${theme.media.lteMedium} {
      margin-bottom: ${theme.spacings.lg};
      text-align: center;
    }
  `}
`;

export const ImgWrap = styled.div`
  ${({ theme }) => css``}
`;

export const Imgfn = styled.img`
  ${({ theme }) => css``}
`;
