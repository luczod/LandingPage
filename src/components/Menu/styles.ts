import styled, { DefaultTheme, css } from "styled-components";
import { Container as SectionContainer } from "../SectionContainer/styles";
import { Title as Heading } from "../Heading/styles";

interface IProps {
  $visible: boolean;
}

const menuVisible = () => css`
  visibility: visible;
  opacity: 1;
`;

export const Container = styled.header<IProps>`
  ${({ theme, $visible }) => css`
    position: fixed;
    z-index: 5;
    top: 0;
    left: 0;
    right: 0;
    transition: all 300ms ease-in-out;
    background: ${theme.colors.white};
    border: 0.1rem solid ${theme.colors.mediumGray};

    > ${SectionContainer} {
      padding-top: 0;
      padding-bottom: 0;
    }

    & ${Heading} {
      margin-top: 0;
      margin-bottom: 0;
    }

    @media ${theme.media.lteMedium} {
      height: 100vh;
      visibility: hidden;
      opacity: 0;
      ${$visible && menuVisible()}

      > ${SectionContainer} {
        height: 100%;
        width: 100%;
        margin: 0;
        padding: 0;
      }

      & ${Heading} {
        padding-bottom: ${theme.spacings.lg};
        display: flex;
        justify-content: center;
      }
    }
  `}
`;

export const MenuContainer = styled.div`
  ${({ theme }) => css`
    display: flex;
    justify-content: space-between;
    align-items: center;
    margin-left: 20rem;
    margin-right: 20rem;

    @media ${theme.media.lteMedium} {
      flex-flow: column nowrap;
      justify-content: flex-start;
      align-items: center;
      text-align: center;
      min-height: 100vh;
      height: 100%;
      overflow-y: auto;
      padding: ${theme.spacings.xxl} 0;
      cursor: pointer;
    }
  `}
`;

export const Btn = styled.button<IProps>`
  ${({ theme, $visible }) => css`
    z-index: 6;
    position: fixed;
    top: 2rem;
    right: 3rem;
    background: ${theme.colors.primaryColor};
    color: ${theme.colors.white};
    border: none;
    display: none;
    pointer-events: ${$visible ? "none" : "all"};

    @media ${theme.media.lteMedium} {
      display: flex;
      align-items: center;
      justify-content: center;
    }

    > svg {
      cursor: pointer;
      padding-top: 2px;
      padding-left: 6px;
      padding-right: 6px;
    }
  `}
`;
