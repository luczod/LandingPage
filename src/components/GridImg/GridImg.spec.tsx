import { screen } from "@testing-library/react";
import { renderTheme } from "../../styles/render-theme";
import { GridImg } from ".";

describe("<GridImg/>", () => {
  it("should render", () => {
    renderTheme(
      <GridImg>
        <h1>Texto</h1>
      </GridImg>
    );
    expect(screen.getByRole("heading", { name: "Texto" })).toBeInTheDocument();
  });
});
