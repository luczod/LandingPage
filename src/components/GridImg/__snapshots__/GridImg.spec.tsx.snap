// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<GridImg/> > should render images in a grid 1`] = `
.c0 {
  background-color: #FFFFFF;
  color: #0A1128;
  border-bottom: 0.2rem solid #DDDDDD;
  min-height: 100vh;
  display: flex;
  align-items: center;
}

.c1 {
  max-width: 120rem;
  margin: 0 auto;
  padding: 2.4rem;
}

.c3 {
  color: #0A1128;
  font-size: 6.4rem;
  text-transform: uppercase;
}

.c5 {
  font-size: 2.4rem;
}

.c2 >.c4 {
  margin-bottom: 3.2rem;
}

.c6 {
  counter-reset: grid-counter;
  display: grid;
  grid-template-columns: repeat(auto-fill, minmax(280px, 1fr));
  gap: 3.2rem;
}

.c7 {
  overflow: hidden;
  border-radius: 1rem;
}

.c8 {
  width: 100%;
  transition: all 300ms ease-in-out;
}

.c8:hover {
  transform: scale(1.2) rotate(10deg);
}

.c9 {
  display: none;
  position: fixed;
  z-index: 1;
  padding-top: 100px;
  left: 0;
  top: 0;
  width: 100%;
  height: 100%;
  background-color: rgba(0 0 0 / 0.3);
  border-radius: 1rem;
}

.c9>span {
  position: fixed;
  cursor: pointer;
  top: 4rem;
  right: 2rem;
  color: #dc143c;
  font-size: 5.6rem;
}

.c9>span::hover {
  color: #FFFFFF;
}

.c9>img {
  margin: auto;
  display: block;
  max-width: 700px;
  object-fit: cover;
  border-radius: 1rem;
}

.c9>div {
  margin: auto;
  display: block;
  width: 80%;
  max-width: 700px;
  text-align: center;
  font-size: 2.4rem;
  color: #FFFFFF;
  padding: 10px 0;
  height: 150px;
}

@media (max-width:768px) {
  .c3 {
    font-size: 4.0rem;
  }
}

@media only screen and (max-width: 768px) {
  .c9>span {
    left: 1rem;
  }
}

@media only screen and (max-width: 768px) {
  .c9>img {
    width: 100%;
    padding: 0 0.8rem;
  }
}

<div>
  <section
    class="c0"
  >
    <div
      class="c1"
    >
      <div
        class="c2"
      >
        <h2
          class="c3"
        >
          My image grid
        </h2>
        <div
          class="c4 c5"
        >
          Atque doloribus nobis laudantium esse ut, non commodi maxime distinctio veritatis unde
        </div>
        <div
          class="c6"
        >
          <div
            class="c7"
          >
            <img
              alt="Teste 1"
              aria-label="item-img"
              class="c8"
              src="https://source.unsplash.com/random/800x800?r=1"
            />
            <div
              aria-label="Modal"
              class="c9"
              id="GridImg-8"
            >
              <img
                alt="Teste 1"
                src="https://source.unsplash.com/random/800x800?r=1"
              />
              <span
                aria-label="Close-img"
              >
                ×
              </span>
              <div>
                Teste 1
              </div>
            </div>
          </div>
          <div
            class="c7"
          >
            <img
              alt="Teste 2"
              aria-label="item-img"
              class="c8"
              src="https://source.unsplash.com/random/800x800?r=2"
            />
            <div
              aria-label="Modal"
              class="c9"
              id="GridImg-9"
            >
              <img
                alt="Teste 2"
                src="https://source.unsplash.com/random/800x800?r=2"
              />
              <span
                aria-label="Close-img"
              >
                ×
              </span>
              <div>
                Teste 2
              </div>
            </div>
          </div>
          <div
            class="c7"
          >
            <img
              alt="Teste 3"
              aria-label="item-img"
              class="c8"
              src="https://source.unsplash.com/random/800x800?r=3"
            />
            <div
              aria-label="Modal"
              class="c9"
              id="GridImg-10"
            >
              <img
                alt="Teste 3"
                src="https://source.unsplash.com/random/800x800?r=3"
              />
              <span
                aria-label="Close-img"
              >
                ×
              </span>
              <div>
                Teste 3
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </section>
</div>
`;
